src/main/java/io/anserini/index/IndexReaderUtils.java

```diff
@@ -280,21 +261,71 @@ public IndexTerm next() {
     };
   }
 
-  public static List<Posting> getPostingsList(IndexReader reader, String termStr)
-      throws IOException, ParseException {
-    EnglishAnalyzer ea = new EnglishAnalyzer(CharArraySet.EMPTY_SET);
-    QueryParser qp = new QueryParser(IndexArgs.CONTENTS, ea);
-    TermQuery q = (TermQuery) qp.parse(termStr);
-    Term t = q.getTerm();
+  /**
+   * Returns the postings list for an unanalyzed term. That is, the method analyzes the term before looking up its
+   * postings list.
+   *
+   * @param reader index reader
+   * @param term unanalyzed term
+   * @return the postings list for an unanalyzed term
+   */
+  public static List<Posting> getPostingsList(IndexReader reader, String term) {
+    return _getPostingsList(reader, AnalyzerUtils.analyze(term).get(0));
+  }
 
-    PostingsEnum postingsEnum = MultiTerms.getTermPostingsEnum(reader, IndexArgs.CONTENTS, t.bytes());
+  /**
+   * Returns the postings list for a term.
+   *
+   * @param reader index reader
+   * @param term term
+   * @param analyze whether or not the method should analyze the term first
+   * @return the postings list for a term
+   */
+  public static List<Posting> getPostingsList(IndexReader reader, String term, boolean analyze) {
+    return _getPostingsList(reader, analyze ? AnalyzerUtils.analyze(term).get(0) : term);
+  }
+
+  /**
+   * Returns the postings list for a term after analysis with a specific analyzer.
+   *
+   * @param reader index reader
+   * @param term term
+   * @param analyzer analyzer
+   * @return the postings list for an unanalyzed term
+   */
+  public static List<Posting> getPostingsList(IndexReader reader, String term, Analyzer analyzer) {
+    return _getPostingsList(reader, AnalyzerUtils.analyze(analyzer, term).get(0));
+  }
+
+  // Internal helper: takes the analyzed form in all cases.
+  private static List<Posting> _getPostingsList(IndexReader reader, String raw) {
```

Comment:
`analyzedTerm` as the parameter name instead of `raw`?

Reply:
Done!